dxvk verb update

bobwya opened the pull request:
@austin987 

I've updated all the dxvk verbs as discussed on IRC. I've also switched over to the Github API (release.json) for pulling the latest release version.

austin987 commented on src/winetricks:
```diff
@@ -7310,12 +7325,11 @@ w_metadata dxvk dlls \
-    w_linkcheck_ignore=1 w_download "https://github.com/doitsujin/dxvk/releases/download/v${dxvk_version}/dxvk-${dxvk_version}.tar.gz"
-    helper_dxvk "dxvk-${dxvk_version}.tar.gz" "d3d10_enabled" "3.19" "1.1.88"
-    unset dxvk_version
+	w_download_to "${W_TMP_EARLY}" "https://api.github.com/repos/doitsujin/dxvk/releases/latest" "" "release.json"
```
Stray tab.

austin987 commented on src/winetricks:
```diff
@@ -7310,12 +7325,11 @@ w_metadata dxvk dlls \
-    helper_dxvk "dxvk-${dxvk_version}.tar.gz" "d3d10_enabled" "3.19" "1.1.88"
-    unset dxvk_version
+	w_download_to "${W_TMP_EARLY}" "https://api.github.com/repos/doitsujin/dxvk/releases/latest" "" "release.json"
+    _W_dxvk_version="$(awk -F':' '{ gsub("[\" ,]",""); if ($1 == "tag_name") {sub("^v","",$2); print $2; exit} }' "${W_TMP_EARLY}/release.json")"
```
Not necessarily a blocker, but this is broken on aria2c, see https://github.com/aria2/aria2/issues/1389.

I was thinking we could make a helper for github API release checks, then have two codepaths, one for curl/wget, one for aria2c (not sure about fetch).

austin987 commented on src/winetricks:
```diff
@@ -6825,13 +6825,11 @@ load_dxsdk_jun2010()
 # $1 - dxvk archive name (required)
 # $2 - dxvk d3d10_enabled / d3d10_disabled (required)
 # $3 - minimum Wine version (optional)
-# $4 - minimum Vulkan API version (optional, requires $3 be set)
```
I'm still not clear on why this needs to go. It doesn't show for everyone, only if it fails the check, which to me is a good thing. Is the check broken in some way? Can we fix it instead?

bobwya replied:
@austin987 

Since I can't get a clear answer from Phillip around the Vulkan version requirements... He talks in terms of required extensions. I gather this is mainly an issue around graphics drivers - that may implement extensions, required for DXVK, but do so on top of older versions of Vulkan (that wouldn't normally support these additional extensions)...

Nobody is going to have a good experience with a recent version of DXVK and Wine 3.x (whatever)... Vulkan 1.1.88 is very out-of-date vs. the very recent driver requirements DXVK imposes on Mesa and the Nvidia driver.

I could suggest Wine and Vulkan versions, based closely matching the Vulkan headers that Wine pulls in and that DXVK pulls in. This would be a "recommended target" (in that DXVK may not use all the features of the Vulkan header version being pulls in) - but at least it would be a meaningful guideline. Rather than the current warning that is displayed and has little basis in reality... 

Thoughts?!
Bob

austin987 replied:
Makes sense, thanks.

bobwya commented:
@austin987 I've reworked and tidied up all the **dxvk** verbs and added support for: [d9vk](https://github.com/Joshua-Ashton/d9vk) - which is heading rapidly towards a useable state (with some quick testing)... **d9vk** supports D3D9 games, by translating D3D9 calls to D3D11, using **dxvk**+Vulkan.

austin987 commented:
Changing verb names breaks computability. You'll want to put a shim, look towards the end of the file. Adding a warning in the shim would probably be good too.

bobwya commented:
@austin987 I've added the **dxvk0*** compatibility shims, as you requested!
